### ISSUE TITLE
有监督学习的损失函数总结

### ISSUE THREAD

**Issue by meton-robean** [meton-robean/ResearchNote#14]
[有监督学习的损失函数总结](https://www.zhihu.com/search?type=content&q=%E6%9C%89%E7%9B%91%E7%9D%A3%E5%AD%A6%E4%B9%A0%E7%9A%84%E6%8D%9F%E5%A4%B1%E5%87%BD%E6%95%B0)

**Comment by meton-robean** [comment]
分类中为什么不用均方差损失？上文在介绍均方差损失的时候讲到**实际上均方差损失假设了误差服从高斯分布，在分类任务下这个假设没办法被满足，因此效果会很差**。为什么是交叉熵损失呢？有两个角度可以解释这个事情，一个角度从最大似然的角度，也就是我们上面的推导；另一个角度是可以用信息论来解释交叉熵损失。通过**最小化交叉熵的角度推导出来的结果和使用最大 化似然得到的结果是一致的。**

**Comment by meton-robean** [comment]
![Selection_011](https://user-images.githubusercontent.com/17881739/67647725-386e9200-f96e-11e9-96cd-5f9610581498.png)
**正是因为HingeLoss的零区域对应的正是非支持向量的普通样本，从而所有的普通样本都不参与最终超平面的决定，这才是支持向量机最大的优势所在，对训练样本数目的依赖大大减少，而且提高了训练效率。**

**Comment by meton-robean** [comment]
![Selection_012](https://user-images.githubusercontent.com/17881739/67648148-d44ccd80-f96f-11e9-8ef2-0d5422ac2d27.png)

**Comment by meton-robean** [comment]
所谓的logistic loss就是 cross entropy: L = -( y1 * log(y1_hat) + y2 * log(y2_hat) ) 在 y1 + y2 = 1时候的化简而已。

**Comment by meton-robean** [comment]
[卷积神经网络系列之softmax，softmax loss和cross entropy的讲解](https://blog.csdn.net/u014380165/article/details/77284921)
![Selection_013](https://user-images.githubusercontent.com/17881739/67649626-ccdbf300-f974-11e9-90b0-253f26323934.png)

**Comment by meton-robean** [comment]
**多分类和多标签分类不是一回事**

**Comment by meton-robean** [comment]
![Selection_014](https://user-images.githubusercontent.com/17881739/67649920-e6316f00-f975-11e9-8e64-2321e03dd8ed.png)

**L1对应拉普拉斯分布，L2对应高斯分布。

L1不可导可以使用Proximal Algorithms或者ADMM来解决。**

**Comment by meton-robean** [comment]
### 神经网络是使用什么如何处理带L1正则化的损失函数的？
作者：Xenophon Tony
链接：https://www.zhihu.com/question/327992018/answer/709912550
来源：知乎
著作权归作者所有。商业转载请联系作者获得授权，非商业转载请注明出处。

首先你说，Lasso不是用coordinate descent或LARS实现的么？小孩子才看算法，成年人都看收敛率，本质上来说你可以用任何优化算法来优化Lasso，包括基于梯度的方法。基于梯度的方法有很多，最常见的便是gradient descent，当问题不再处处可导时，可以采用gradient descent的拓展算法subgradient descent。但一般来说，subgradient descent常常令人头疼，其实它并不是一个保证descent的算法，只不过是作为gd的超集，沿用了descent的这个叫法。事实上subgradient descent的收敛率也非常差，除非迫不得已，不会用它来优化Lasso这种非平滑的函数，也很难令Lasso得到稀疏解。有意思的是，Tensorflow里面实现的甚至不是subgradient descent算法，而是忽略所有不可导的地方，也就是说但凡你遇到了这样一个不可导的坏点，它可以永远卡在那里，因为没有梯度也不会更新。这其实也合理，当网络中有大量参数的时候，相较正牌subgradient descent而言，这是保证效率也不失精准性的一种妥协。但是记住，这是相较subgradient descent而言，基于梯度的方法难道没有更好更合理的了么？有的，那也就是优化Lasso的大杀器proximal gradient descent（Lasso优化中大名鼎鼎的ISTA），抛开理论的部分不谈，对于Lasso而言，proximal方法相当把原问题拆解成了两个子问题，首先更新平滑部分的参数（L1以外的部分），拿着更新好的结果，跑一遍proximal operator（soft threshold函数）让它也尽量更接近满足L1的条件，处理过后的才是新的参数。ISTA是Lasso的随机优化中非常经典的一个算法（当然从收敛效率上来说，coordinate descent要高很多，然而即便是CD他的每一步其实做的也是proximal update，又扯远了…）那最有意思的地方来了，神经网络中，有没有这种神乎其神的proximal方法，可以在普通的gradient descent基础上，提高算法的收敛率呢？有的，比如说这一篇Data-Driven Sparse Structure Selection for Deep Neural Networks，就是一篇利用Accelerated Proximal Gradient方法来产生网络的稀疏结构从而对网络进行剪枝的有趣工作～(⁎⁍̴̛ᴗ⁍̴̛⁎)
